libs/server-sdk/src/data_store/dependency_tracker.cpp

```diff
@@ -0,0 +1,186 @@
+#include "dependency_tracker.hpp"
+
+namespace launchdarkly::server_side::data_store {
+
+DependencySet::DependencySet()
+    : data_{
+          TaggedData<std::set<std::string>>(DataKind::kFlag),
+          TaggedData<std::set<std::string>>(DataKind::kSegment),
+      } {}
+
+void DependencySet::Set(DataKind kind, std::string key) {
+    data_[static_cast<std::size_t>(kind)].Data().emplace(std::move(key));
+}
+
+void DependencySet::Remove(DataKind kind, std::string const& key) {
+    data_[static_cast<std::size_t>(kind)].Data().erase(key);
+}
+
+bool DependencySet::Contains(DataKind kind, std::string const& key) {
```

Comment:
Should be const?

Reply:
I made it non-const because making it const was hell. 
Because then `.Data()` needs a const equivalent. Turns into rust.

Reply:
I can see if I can make it less terrible.